AsyncDisplayKit/Private/_ASPendingState.mm

```diff
@@ -660,144 +667,175 @@ - (void)applyToView:(UIView *)view
 
-  if (_flags.setBorderWidth)
+  if (flags.setBorderWidth)
     layer.borderWidth = borderWidth;
 
-  if (_flags.setBorderColor)
+  if (flags.setBorderColor)
     layer.borderColor = borderColor;
 
-  if (_flags.setAutoresizingMask)
+  if (flags.setAutoresizingMask)
     view.autoresizingMask = autoresizingMask;
 
-  if (_flags.setAutoresizesSubviews)
+  if (flags.setAutoresizesSubviews)
     view.autoresizesSubviews = autoresizesSubviews;
 
-  if (_flags.setNeedsDisplayOnBoundsChange)
+  if (flags.setNeedsDisplayOnBoundsChange)
     layer.needsDisplayOnBoundsChange = needsDisplayOnBoundsChange;
 
-  if (_flags.setAllowsEdgeAntialiasing)
+  if (flags.setAllowsEdgeAntialiasing)
     layer.allowsEdgeAntialiasing = allowsEdgeAntialiasing;
 
-  if (_flags.setEdgeAntialiasingMask)
+  if (flags.setEdgeAntialiasingMask)
     layer.edgeAntialiasingMask = edgeAntialiasingMask;
 
-  if (_flags.needsDisplay)
-    [view setNeedsDisplay];
-
-  if (_flags.needsLayout)
+  if (flags.needsLayout)
     [view setNeedsLayout];
 
-  if (_flags.setAsyncTransactionContainer)
+  if (flags.setAsyncTransactionContainer)
     view.asyncdisplaykit_asyncTransactionContainer = asyncTransactionContainer;
 
-  if (_flags.setOpaque)
+  if (flags.setOpaque)
     ASDisplayNodeAssert(view.layer.opaque == opaque, @"Didn't set opaque as desired");
 
-  if (_flags.setIsAccessibilityElement)
+  if (flags.setIsAccessibilityElement)
     view.isAccessibilityElement = isAccessibilityElement;
 
-  if (_flags.setAccessibilityLabel)
+  if (flags.setAccessibilityLabel)
     view.accessibilityLabel = accessibilityLabel;
 
-  if (_flags.setAccessibilityHint)
+  if (flags.setAccessibilityHint)
     view.accessibilityHint = accessibilityHint;
 
-  if (_flags.setAccessibilityValue)
+  if (flags.setAccessibilityValue)
     view.accessibilityValue = accessibilityValue;
 
-  if (_flags.setAccessibilityTraits)
+  if (flags.setAccessibilityTraits)
     view.accessibilityTraits = accessibilityTraits;
 
-  if (_flags.setAccessibilityFrame)
+  if (flags.setAccessibilityFrame)
     view.accessibilityFrame = accessibilityFrame;
 
-  if (_flags.setAccessibilityLanguage)
+  if (flags.setAccessibilityLanguage)
     view.accessibilityLanguage = accessibilityLanguage;
 
-  if (_flags.setAccessibilityElementsHidden)
+  if (flags.setAccessibilityElementsHidden)
     view.accessibilityElementsHidden = accessibilityElementsHidden;
 
-  if (_flags.setAccessibilityViewIsModal)
+  if (flags.setAccessibilityViewIsModal)
     view.accessibilityViewIsModal = accessibilityViewIsModal;
 
-  if (_flags.setShouldGroupAccessibilityChildren)
+  if (flags.setShouldGroupAccessibilityChildren)
     view.shouldGroupAccessibilityChildren = shouldGroupAccessibilityChildren;
 
-  if (_flags.setAccessibilityIdentifier)
+  if (flags.setAccessibilityIdentifier)
     view.accessibilityIdentifier = accessibilityIdentifier;
+
+  // FIXME: How should we reconcile order-of-operations between setting frame, bounds, position?
```

Comment:
Layer-backing is very much a first-class feature of the framework. Why is the behavior of this method different than the layer one? Can we consolidate and share the code, and thereby ensure that any subtleties of the behavior that we do your support are reliably consistent across view and layer back nodes?